vault/resource_userpass_user.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func userpassUserResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"backend": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Auth backend to which user will be configured.",
+			ForceNew:    true,
+			Default:     "userpass",
+			// standardise on no beginning or trailing slashes
+			StateFunc: func(v interface{}) string {
+				return strings.Trim(v.(string), "/")
+			},
+		},
+		"username": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "Username part of userpass.",
+			ValidateFunc: provider.ValidateStringSlug,
+		},
+		"password": {
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Password part of userpass.",
+			Sensitive:   true,
+		},
+	}
+
+	addTokenFields(fields, &addTokenFieldsConfig{})
+
+	return &schema.Resource{
+		Create: userpassUserCreate,
```

Comment:
Could we update these to use the non-deprecated SDK functions like `CreateContext`, `ReadContext` instead? Examples for this are [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kv_secret_v2.go#L36)